nextflow_schema.json

```diff
@@ -479,6 +479,82 @@
                 }
             },
             "fa_icon": "fas fa-layer-group"
+        },
+        "bam_filtering": {
+            "title": "BAM Filtering",
+            "type": "object",
+            "description": "Options related to length, quality, and map status filtering of reads.",
+            "default": "",
+            "properties": {
+                "run_bamfiltering": {
+                    "type": "boolean",
+                    "description": "Turn on BAM filtering of mapped BAM files. By default, only mapped reads retained.",
+                    "fa_icon": "fas fa-power-off",
+                    "help_text": "Turns on the filtering subworkflow for mapped BAM files coming out of the read alignment step. Filtering includes removal of unmapped reads, length filtering, and mapping quality filtering.\n\nWhen turning on bam filtering, by default the only filter mapped/unmapped filter is activated, thus only mapped reads are retained for downstream analyses. See `--bamfiltering_retainunmappedgenomicbam` to retain unmapped reads, if only length or quality filtering preferred.\n\nNote this subworkflow can also be activated if `--run_metagenomic_screening` is supplied."
+                },
+                "bamfiltering_minreadlength": {
+                    "type": "integer",
+                    "default": 0,
+                    "description": "Specify the minimum read length mapped reads should have for downstream genomic analysis.",
+                    "help_text": "You can use this to remove mapped reads that fall below a certain length after mapping.\n\nThis can be useful to get more realistic 'endogenous DNA' or 'on target read' percentages, when most of your reads are very short (e.g. in single-stranded libraries) that would otherwise be discarded during a length filtering step at the earlier read clipping and merging  stage (whereby in the latter case, removing short reads prior mapping would make an artificially small denominator for a typical endogenous DNA calculation and inflate on-target statistics). Note in this context you should not perform mapping quality filtering nor discarding of unmapped reads to ensure a correct denominator of all reads, for the endogenous DNA calculation.\n\nBy specifying 0, no length filtering is performed.\n\nNote that by default the output BAM files of this step are _not_ stored in the results directory (as it is assumed that deduplicated BAM files are preferred). See `--bamfiltering_savefilteredbams` if you wish to save these.\n\n> Modifies tool parameter(s):\n> - filter_bam_fragment_length.py: `-l`",
```

Comment:
```suggestion
                    "help_text": "You can use this to remove mapped reads that fall below a certain length after mapping.\n\nThis can be useful to get more realistic 'endogenous DNA' or 'on target read' percentages, when most of your reads are very short (e.g. in single-stranded libraries) that would otherwise be discarded during a length filtering step at the earlier read clipping and merging  stage (whereby in the latter case, removing short reads prior mapping would make an artificially small denominator for a typical endogenous DNA calculation and inflate on-target statistics). Note in this context you should not perform mapping quality filtering nor discard unmapped reads, to ensure a correct denominator of all reads for the endogenous DNA calculation.\n\nBy specifying `0`, no length filtering is performed.\n\nNote that by default the output BAM files of this step are _not_ stored in the results directory (as it is assumed that deduplicated BAM files are preferred). See `--bamfiltering_savefilteredbams` if you wish to save these.\n\n> Modifies tool parameter(s):\n> - filter_bam_fragment_length.py: `-l`",
```